presentation/src/main/java/com/architect/coders/mu8/categories/CategoriesActivity.kt

```diff
@@ -34,22 +41,25 @@ class MainActivity : AppCompatActivity() {
 
         recycler = findViewById(R.id.marvel_list)
 
-        viewModel = ViewModelProviders.of(
-            this,
-            CategoriesViewModelFactory(CategoriesRepository())
-        )[CategoriesViewModel::class.java]
-
+        viewModel = getViewModel { CategoriesViewModel(CategoriesRepository()) }
         viewModel.model.observe(this, Observer(::updateUi))
     }
 
     private fun updateUi(model: UiModel) {
         when (model) {
             is UiModel.Content -> {
-                adapter = CategoriesAdapter(viewModel::onMoviewClicked, model.categories)
+                adapter = CategoriesAdapter(viewModel::onCategoryClicked, model.categories)
                 recycler.layoutManager = LinearLayoutManager(this)
                 recycler.adapter = adapter
             }
-            is UiModel.Navigation -> Toast.makeText(this, model.category.title, Toast.LENGTH_SHORT).show()
+            is UiModel.Navigation -> {
+                when (model.categoryName) {
+                    CHARACTERS -> startActivity<CharactersActivity> {}
+                    COMICS -> Toast.makeText(this, COMICS, Toast.LENGTH_SHORT).show()
```

Comment:
use  `COMICS,EVENTS -> Toast.makeText(this, EVENTS, Toast.LENGTH_SHORT).show()` in a same code line.

Reply:
That won't allow me to show the toast with the right category, if you can see on your snippet you'll show the toast for both categories with `EVENTS`, what I can do is to reuse the Toast in a method for now.